odoo/orm/fields_properties.py

```diff
@@ -29,6 +29,54 @@ def check_property_field_value_name(property_name):
     if not (0 < len(property_name) <= 512) or not regex_alphanumeric.match(property_name):
         raise ValueError(f"Wrong property field value name {property_name!r}.")
 
+class Property(dict):
+    """Allow to be used in `convert_to_record` to make the verification only if we try accessing the property value."""
+
+    def __init__(self, values, field, record):
+        super().__init__(values)
+        self.record = record
+        self.field = field
+
+    def __getitem__(self, property_name):
+        """Will make the verification."""
+        if not self.record:
+            return False
+
+        records = self.record | self.record.browse(islice(self.record._prefetch_ids, 1000))
+        records = records.filtered(lambda r: r[self.field.definition_record] == self.record[self.field.definition_record])
+
+        values = self.field.convert_to_read(records.env.cache.get(records, self.field), self.record, use_display_name=False)
+        prop = next((p for p in values if p['name'] == property_name), False)
+        if not prop:
+            # We need to know if it's a relational property defined on a different definition record
+            definition = self.record.get_property_definition(f"{self.field.name}.{property_name}")
+            if not definition:
+                raise KeyError(self.record.env._("Property doesn't exist"))
+
+            prop = definition
+
+        if prop.get('type') == 'many2one' and prop.get('comodel'):
+            prefetch_ids = [prop['default']] if prop.get('default') else []
+            return self.record.env[prop.get('comodel')].browse(prop.get('value')).with_prefetch()
+
+        if prop.get('type') == 'many2many' and prop.get('comodel'):
+            prefetch_ids = list(prop['default']) if prop.get('default') else []
+            return self.record.env[prop.get('comodel')].browse(prop.get('value')).with_prefetch(prefetch_ids)
```

Comment:
Please explain.  The value is a recordset with ids like (42, 43, 45) and with unrelated prefetch ids like (1, 2, 3).  A loop on the recordset will fetch each record separately.  That call to `with_prefetch()` is a _counter_-optimization.

I suggest to simply ignore prefetching at the moment, we can improve this later.